tests/src/test/java/alluxio/shell/command/CpCommandTest.java

```diff
@@ -0,0 +1,132 @@
+import org.apache.commons.io.IOUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+
+/**
+ * Tests for cp command.
+ */
+public final class CpCommandTest extends AbstractAlluxioShellTest {
+
+  /**
+   * Tests recursively copying a directory.
+   */
+  @Test
+  public void copyDirTest() throws Exception {
+    AlluxioShellUtilsTest.resetFileHierarchy(mFileSystem);
+    int ret = mFsShell.run("cp", "/testWildCards", "/copy");
+    Assert.assertEquals(0, ret);
+    Assert.assertTrue(mFileSystem.exists(new AlluxioURI("/copy")));
+    Assert.assertTrue(mFileSystem.exists(new AlluxioURI("/copy/bar")));
+    Assert.assertTrue(mFileSystem.exists(new AlluxioURI("/copy/bar/foobar3")));
+    Assert.assertTrue(mFileSystem.exists(new AlluxioURI("/copy/foo")));
+    Assert.assertTrue(mFileSystem.exists(new AlluxioURI("/copy/foo/foobar1")));
+    Assert.assertTrue(mFileSystem.exists(new AlluxioURI("/copy/foo/foobar2")));
+    Assert.assertTrue(mFileSystem.exists(new AlluxioURI("/copy/foobar4")));
+
+    Assert.assertTrue(
+        equals(new AlluxioURI("/copy/bar/foobar3"), new AlluxioURI("/testWildCards/bar/foobar3")));
+    Assert.assertTrue(
+        equals(new AlluxioURI("/copy/foo/foobar1"), new AlluxioURI("/testWildCards/foo/foobar1")));
+    Assert.assertTrue(
+        equals(new AlluxioURI("/copy/foo/foobar2"), new AlluxioURI("/testWildCards/foo/foobar2")));
+    Assert.assertTrue(
+        equals(new AlluxioURI("/copy/foobar4"), new AlluxioURI("/testWildCards/foobar4")));
+  }
+
+  /**
+   * Tests copying a list of files specified through a wildcard expression.
+   */
+  @Test
+  public void copyWildcardDirTest() throws Exception {
+    AlluxioShellUtilsTest.resetFileHierarchy(mFileSystem);
+    int ret = mFsShell.run("cp", "/testWildCards/*/foo*", "/copy");
+    Assert.assertEquals(0, ret);
+    Assert.assertTrue(mFileSystem.exists(new AlluxioURI("/copy")));
+    Assert.assertTrue(mFileSystem.exists(new AlluxioURI("/copy/foobar1")));
+    Assert.assertTrue(mFileSystem.exists(new AlluxioURI("/copy/foobar2")));
+    Assert.assertTrue(mFileSystem.exists(new AlluxioURI("/copy/foobar3")));
+    Assert.assertFalse(mFileSystem.exists(new AlluxioURI("/copy/foobar4")));
+
+    Assert.assertTrue(
+        equals(new AlluxioURI("/copy/foobar1"), new AlluxioURI("/testWildCards/foo/foobar1")));
+    Assert.assertTrue(
+        equals(new AlluxioURI("/copy/foobar2"), new AlluxioURI("/testWildCards/foo/foobar2")));
+    Assert.assertTrue(
+        equals(new AlluxioURI("/copy/foobar3"), new AlluxioURI("/testWildCards/bar/foobar3")));
+  }
+
+  /**
+   * Tests invalid input arguments.
+   */
+  @Test
+  public void copyWildcardNotDirTest() throws Exception {
+    AlluxioShellUtilsTest.resetFileHierarchy(mFileSystem);
+    int ret;
+    ret = mFsShell.run("cp", "/testWildCards", "/testWildCards/foobar4");
+    Assert.assertEquals(-1, ret);
+    ret = mFsShell.run("cp", "/testWildCards/*", "/testWildCards/foobar4");
```

Comment:
it will create a copy of /testWildCards (before the operation) under /testWildCards/testWildCards